### PR TITLE
Add utility to get hostname

### PR DESCRIPTION
**Pull request by JosephusPaye**
This adds a utility to get the system hostname, which uses the env variable `ROBOT_HOSTNAME` if set, otherwise uses `gethostname()`. Also updates everywhere we use `gethostname()` to use the utility.

### CODE REVIEW

**Review by TrentHouliston — CHANGES_REQUESTED** [review]
Need to fix up Nuclear Network module as well

**Comment by JosephusPaye** [comment]
> Need to fix up Nuclear Network module as well

Where is the hostname used in there?

File: shared/utility/support/hostname.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include "hostname.hpp"
+
+namespace utility::support {
+
+    std::string getHostname() {
+        char* hostname = std::getenv("ROBOT_HOSTNAME");
+
+        if (hostname == nullptr) {
+            char buffer[255];
+            gethostname(buffer, 255);
```
Comment (Bidski) [inline shared/utility/support/hostname.cpp:10]: Pedanticness I guess
```suggestion
            ::gethostname(buffer, 255);
```

File: shared/utility/support/hostname.hpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#ifndef UTILITY_SUPPORT_HOSTNAME_HPP
+#define UTILITY_SUPPORT_HOSTNAME_HPP
+
+#include <string>
+#include <unistd.h>
```
Comment (Bidski) [inline shared/utility/support/hostname.hpp:5]:
```suggestion
extern "C" {
#include <unistd.h>
}
```

File: shared/utility/support/hostname.hpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace utility::support {
+
+    std::string getHostname();
```
Comment (Bidski) [inline shared/utility/support/hostname.hpp:9]: I feel like you could just move the entire function into here
```suggestion
    inline std::string getHostname() {
        char* hostname = std::getenv("ROBOT_HOSTNAME");

        if (hostname == nullptr) {
            char buffer[255];
            ::gethostname(buffer, 255);    
            hostname = buffer;
        }

        return std::string(hostname);            
    }
```

**Review by Bidski — CHANGES_REQUESTED** [review]

**Comment by Bidski** [comment]
> > Need to fix up Nuclear Network module as well
> 
> Where is the hostname used in there?

Apparently here
https://github.com/NUbots/NUbots/blob/master/module/network/NUClearNet/src/NUClearNet.cpp#L30

File: shared/utility/support/hostname.hpp (hunk 3 above)
Comment (JosephusPaye) [reply shared/utility/support/hostname.hpp:9]: Yeah, was wondering that too

File: module/network/NUClearNet/src/NUClearNet.cpp [hunk 4]
```diff
@@ -27,7 +29,8 @@ namespace module::network {
 
         on<Configuration>("NUClearNet.yaml").then([this](const Configuration& config) {
             auto netConfig              = std::make_unique<NUClear::message::NetworkConfiguration>();
-            netConfig->name             = config["name"].as<std::string>();
+            std::string name            = config["name"].as<std::string>();
+            netConfig->name             = name.length() == 0 ? utility::support::getHostname() : name;
```
Comment (Bidski) [inline module/network/NUClearNet/src/NUClearNet.cpp:33]:
```suggestion
            netConfig->name             = name.empty() ? utility::support::getHostname() : name;
```

**Review by Bidski — CHANGES_REQUESTED** [review]

**Review by Bidski — APPROVED** [review]
Sure, if it works

**Comment by JosephusPaye** [comment]
Tested and it works!

**Review by TrentHouliston — APPROVED** [review]

**Pull request merged**